### ISSUE TITLE
Verify Make Field Readonly is not reported for assignment to this

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#30262]
A test should be added to verify "Make Field Readonly" is not reported if an assignment to `this` occurs outside the constructor of a struct.

For example, in the following code, no suggestion should be provided to mark `S.Value` as `readonly`.

```csharp
struct S
{
  private int Value;

  void Method() {
    this = default;
  }
}
```

**Comment by CyrusNajmabadi** [comment]
This work was done.
